### PR TITLE
fix MGDCTRS-1548: Azure Storage Blob changefeed source connector

### PR DESCRIPTION
**Pull request by valdar**
the connector produce one record per event
the connector produced data are in json format
the connector add additional info as record headers

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            cnt.start()
+
+        when:
+            bc.upload(BinaryData.fromString(payload))
+
+        then:
+            kafka.poll(group, topic).any {
+                log.info("RECORD = {}", it.value())
+                return it.value() != null
+                        && !it.value().isEmpty()
```
Comment (rinaldodev) [inline cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy:54]: won't we validate the value?

**Review by rinaldodev — CHANGES_REQUESTED** [review]

File: cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy (hunk 1 above)
Comment (valdar) [reply cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy:54]: The value has nothing to do with what we send to the blobl it is something of this form:
```json
{  "api": "DeleteBlob",
    "clientRequestId": "9c2fbc6a-2f7b-4d64-84b5-d593e1a63881",
    "requestId": "1049d8d6-901e-0011-298f-100dd2000000",
    "contentType": "application/octetstream",
    "contentLength":36,
    "blobType":"BlockBlob",
    "contentOffset": null,
    "destinationUrl":null,
    "sourceUrl":null,
    "blobUrl":"https://tnbblob.blob.core.windows.net/engblob/f741f282-51b1-45dd-9826ffcb2a1900ba",
    "recursive":false,
    "sequencer":"000000000000000000000000000085A700000000018617ad",
    "etag":"0x8DADEA719BC0A02"
}
```
I am not sure how to validate it :confused: . Any ideas is welcome.

Comment (valdar) [reply cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy]: I have added a sort of value validation: that is a json, that has a `blobUrl` property.

Comment (rinaldodev) [reply cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy]: This event seem to be a DeleteBlob event, but what you seem to be doing is writing to the Blob. Are we validating the correct event?

Comment (valdar) [reply cos-fleet-catalog-connectors-it/azure/storage-blob-changefeed-it/src/test/groovy/org/bf2/cos/connector/camel/it/azure/storage/blob/AzureStorageBlobChangefeedConnectorIT.groovy]: I picked a random one from some logs I went through while developing it. It was just an example to show that almost nothing in the event is correlated with the content we send.

**Review by rinaldodev — APPROVED** [review]
After talking to Andrea we understand this is not final work and future changes will be done in the Camel component and then to the connector. Currently this is the best we can do.

**Review by rinaldodev — APPROVED** [review]

**Pull request merged**